src/EditorFeatures/CSharp/QuickInfo/SyntacticQuickInfoProvider.cs

```diff
@@ -76,8 +85,53 @@ internal class SyntacticQuickInfoProvider : AbstractQuickInfoProvider
                 return null;
             }
 
-            var span = new SnapshotSpan(textSnapshot, Span.FromBounds(parent.SpanStart, openBrace.Span.End));
+            var span = new SnapshotSpan(textSnapshot, Span.FromBounds(spanStart, spanEnd));
             return this.CreateElisionBufferDeferredContent(span);
         }
+
+        private static void MarkInterestedSpanNearbyScopeBlock(SyntaxNode block, SyntaxToken openBrace, ref int spanStart, ref int spanEnd)
+        {
+            SyntaxTrivia nearbyComment;
+
+            var searchListAbove = openBrace.LeadingTrivia.Reverse();
+            if (FindFurthestNearbyComment(ref searchListAbove, out nearbyComment))
+            {
+                spanStart = nearbyComment.SpanStart;
+                return;
+            }
+
+            var nextToken = block.FindToken(openBrace.FullSpan.End);
+            var searchListBelow = nextToken.LeadingTrivia;
+            if (FindFurthestNearbyComment(ref searchListBelow, out nearbyComment))
+            {
+                spanEnd = nearbyComment.Span.End;
+                return;
+            }
+        }
+
+        private static bool FindFurthestNearbyComment<T>(ref T triviaSearchList, out SyntaxTrivia nearbyTrivia)
```

Comment:
@CyrusNajmabadi Yes, as you described it's an attempt to avoid boxing and copy. Some say it can be annotated with `[MethodImpl(MethodImplOptions.AggressiveInlining)]` to further avoid method call overhead, but I don't think it need this far here.

Sure, renamed.